02-setup-runs.Rmd

```diff
@@ -46,6 +46,132 @@ file.copy(file.path("templates", "workflow_template.R"),
           file.path(wd, "workflow.R"))
 ```
 
+# 5. Edit css/pss files
+
+
+Copy to `data/sites/mandifore/` to edit
+
+```{r}
+file.copy(
+  from = file.path("/data/input", s$cohort_filename),
+  to = "/data/sites/mandifore",
+  recursive = TRUE,
+  copy.mode = FALSE
+)
+```
+
+Read into R
+
+```{r}
+pss_file <- list.files(
+    file.path("/data/sites/mandifore", s$cohort_filename),
+    pattern = ".pss$",
+    full.names = TRUE
+  )
+pss <- read_table(pss_file)
+
+css_file <- list.files(
+    file.path("/data/sites/mandifore", s$cohort_filename),
+    pattern = ".css$",
+    full.names = TRUE
+  )
+css <- read_table(css_file)
+```
+
+
+## pss
+
+Check if colnames are correct
+
+```{r}
+waldo::compare(colnames(pss), c("time", "patch", "trk", "age", "area", "water", "fsc", "stsc", "stsl", "ssc", "lai", "msn", "fsn", "nep", "gpp", "rh"))
+```
+
+Looks like we need to:
+1) remove `site` column
+2) `psc` was an old dummy variable, `lai` is a new dummy variable, so just rename
+3) add dummy columns `nep`, `gpp`, and `rh`
+
+
+```{r}
+pss <- pss |> 
+  select(-site, lai = psc) |> 
+  add_column(nep = 0, gpp = 0, rh = 0)
+
+waldo::compare(colnames(pss), c("time", "patch", "trk", "age", "area", "water", "fsc", "stsc", "stsl", "ssc", "lai", "msn", "fsn", "nep", "gpp", "rh"))
+```
+
+#### Remove empty patches
+
+I'm going to try removing patches that don't exist in the css file
+
+```{r}
+pss <- pss |> 
+  filter(patch %in% unique(css$patch))
```

Comment:
>One thing that might be of use in these initial conditions files is that the area column in the pss files are supposed to sum to 1.

--@KristinaRiemer

Reply:
I may need to re-scale the area column after removing empty patches (if that is indeed what we continue to do)